File: PAT/Basic Level/1032.cpp
#include <iostream>

#include <map>

using std::cin;
using std::cout;
using std::endl;
using std::map;

int main()
{
	int n;
	cin >> n;

	map<int, int> schools;

	int maxscore = 0;
	int maxID = 0;

	for(int i = 0; i < n; i++)
	{
		int ID, score;
		cin >> ID >> score;

		schools[ID] += score;
		if(schools[ID] > maxscore)
		{
			maxscore = schools[ID];
			maxID = ID;
		}
	}
	cout << maxID << ' ' << maxscore << endl;

	return 0;
}

File: PAT/Basic Level/1032.py
N = int(raw_input())

schools = {}

highID = ''
highscore = 0

for i in range(0, N):
    temp = raw_input().split()
    ID = temp[0]
    score = int(temp[1])
    if schools.has_key(ID):
        schools[ID] += score
    else:
    schools[ID] = score
    if schools[ID] > highscore:
        highID = ID
        highscore = schools[ID]

print "%s %d" %(highID, highscore)
